src/home_robot/home_robot/agent/ovmm_agent/ovmm_agent.py

```diff
@@ -50,65 +71,121 @@ def reset_vectorized_for_env(self, e: int):
+        fall_steps = 20
+        num_turns = np.round(90 / turn_angle)
+        forward_and_turn_steps = forward_steps + num_turns
+        if place_step <= forward_steps:
+            # for experimentation (TODO: Remove. ideally nav should drop us close)
+            action = DiscreteNavigationAction.MOVE_FORWARD
+        elif place_step <= forward_and_turn_steps:
+            # first orient
+            action = DiscreteNavigationAction.TURN_LEFT
+        elif place_step == forward_and_turn_steps + 1:
+            action = DiscreteNavigationAction.FACE_ARM
+        elif place_step == forward_and_turn_steps + 2:
+            action = DiscreteNavigationAction.EXTEND_ARM
+        elif place_step == forward_and_turn_steps + 3:
+            # desnap to drop the object
+            action = DiscreteNavigationAction.DESNAP_OBJECT
+        elif place_step <= forward_and_turn_steps + 3 + fall_steps:
+            # allow the object to come to rest
+            action = DiscreteNavigationAction.EMPTY_ACTION
+        elif place_step == forward_and_turn_steps + fall_steps + 4:
+            action = DiscreteNavigationAction.STOP
+        return action
+
+    def act(self, obs: Observations) -> Tuple[DiscreteNavigationAction, Dict[str, Any]]:
+        """State machine"""
+        vis_inputs = self._get_vis_inputs(obs)
+        turn_angle = self.config.ENVIRONMENT.turn_angle
+
+        self.timesteps[0] += 1
+        if self.states[0] == Skill.NAV_TO_OBJ:
+            if self.skip_nav_to_obj:
+                self._switch_to_next_skill(e=0)
+            elif self.config.AGENT.SKILLS.NAV_TO_OBJ.type == "modular":
+                return self._modular_nav(obs)
+            else:
+                raise NotImplementedError
+        if self.states[0] == Skill.ORIENT_OBJ:
+            num_turns = np.round(90 / turn_angle)
+            orient_step = self.timesteps[0] - self.orient_start_step[0]
+            if self.skip_orient_obj:
+                self._switch_to_next_skill(e=0)
+            elif orient_step <= num_turns:
+                return DiscreteNavigationAction.TURN_LEFT, vis_inputs
+            elif orient_step == num_turns + 1:
+                self._switch_to_next_skill(e=0)
+                return DiscreteNavigationAction.FACE_ARM, vis_inputs
+        if self.states[0] == Skill.PICK:
+            if self.skip_pick:
+                self._switch_to_next_skill(e=0)
+            elif self.is_pick_done[0]:
+                self._switch_to_next_skill(e=0)
+                self.is_pick_done[0] = 0
+                return DiscreteNavigationAction.RESET_JOINTS, vis_inputs
+            elif self.config.AGENT.SKILLS.PICK.type == "gaze":
+                action, term = self.gaze_agent.act(obs)
+                if term:
+                    action["grip_action"] = [1]  # grasp the object when gaze is done
+                    self.is_pick_done[0] = 1
+                return action, vis_inputs
+            elif self.config.AGENT.SKILLS.PICK.type == "hardcoded":
```

Comment:
We need to move to having a single OVMM agent in sim and real - should coordinate on changes today

Reply:
Yes, now that I played a bit with the real-world hierarchical policy, I feel this can be done. I will work towards this.

Reply:
different PR though! this one is almost done